### PR TITLE
Implement database integration test harness

### PR DESCRIPTION
**Pull request by rhmdnd**
This commit adds a utility script to stand up a postgres container so we
can run tests against it. It also adds skeleton integration test file
that establishes a connection to the database. Future patches will build
out the tests.

### CODE REVIEW

**Comment by rhmdnd** [comment]
I'm unable to reproduce this locally, but the postgres container is failing to start:

```
 Health status: starting...
Health status: starting...
Health status: starting...
Failed to wait for pgsql container to come up
time="2022-06-16T21:30:38Z" level=error msg="Error removing timer for container cff81e19a4a652e7f3356bdda3daee27130beb66bc1b3fab356ec026e23095b6 healthcheck: unable to get systemd connection to remove healthchecks: dial unix /tmp/podman-run-1001/systemd/private: connect: no such file or directory"
postgres
time="2022-06-16T21:30:38Z" level=error msg="Error removing timer for container cff81e19a4a652e7f3356bdda3daee27130beb66bc1b3fab356ec026e23095b6 healthcheck: unable to get systemd connection to remove healthchecks: dial unix /tmp/podman-run-1001/systemd/private: connect: no such file or directory"
cff81e19a4a652e7f3356bdda3daee27130beb66bc1b3fab356ec026e23095b6
make: *** [Makefile:36: test-database-integration] Error 1
```

File: tests/integration/integration_test.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+func setUpDatabase(t *testing.T) *sql.DB {
+	t.Helper()
+	// Generlize this so that it can be used to connect to any Postgres
+	// database to run tests.
+	connStr := "host=localhost user=dbadmin dbname=compliance password=secret port=5432 sslmode=disable"
+
+	db, err := sql.Open("postgres", connStr)
+	if err != nil {
+		msg := fmt.Sprintf("Unable to initialize connection to test database: %s", err)
+		t.Skip(msg)
```
Comment (rhmdnd) [inline tests/integration/integration_test.go:20]: We skip these tests if we can't establish a connection to the database. I wouldn't be opposed to failing here either, but skipping allows us to continue invoking all tests using `go test -v ./...` which is what we rely on in openshift/release. I'm open to other ways of doing this.

Comment (rhmdnd) [reply tests/integration/integration_test.go:20]: The only tests that utilize this connection are the integration tests in this module. Everything else at this point is considered unit test territory.

**Review by Vincent056 — APPROVED** [review]
/lgtm

**Pull request merged**